### ISSUE TITLE
Signature verification spotlight bans legitimate requests

### ISSUE THREAD

**Issue by ThibG** [tootsuite/mastodon#9323]
A stoplight was added to blacklist sources with failing signature verifications: https://github.com/tootsuite/mastodon/blob/master/app/controllers/concerns/signature_verification.rb#L46-L49

However, this catches any exception, including account name validation as the following:
```
Switching source:185.34.33.17 from green to red because ActiveRecord::RecordInvalid Validation failed: Username only letters, numbers and underscores
```

This is especially bad, as this means that an instance allowing non-Mastodon-compliant names may get banned from a Mastodon instance, stopping accounts there with Mastodon-compliant names from having their messages processed.

**Comment by Gargron** [comment]
>Username only letters, numbers and underscores

Why is this message so gramatically wrong anyway, is it composited from "username" and "only letters ..." separately?

Anyway, I guess we can exempt some exceptions from affecting stoplight and allow dashes in remote usernames.

**Comment by Gargron** [comment]
Fixed by #9617
